File: Prime_Check.c
#include <stdbool.h>
#include <stdio.h>

bool is_prime(int number)
{
    int divisor;

    if (number < 2) {
        return false;
    }
    if (number == 2) {
        return true;
    }
    if (number % 2 == 0) {
        return false;
    }

    for (divisor = 3; divisor <= number / divisor; divisor += 2) {
        if (number % divisor == 0) {
            return false;
        }
    }
    return true;
}

#ifndef PRIME_CHECK_NO_MAIN
int main(void)
{
    int number;

    printf("Enter an integer: ");
    if (scanf("%d", &number) != 1) {
        fprintf(stderr, "Error: please enter a valid integer.\n");
        return 1;
    }

    printf("%d is %s.\n", number, is_prime(number) ? "prime" : "not prime");
    return 0;
}
#endif

File: Prime_Check.py
"""Prime-number checker with a small interactive command-line interface."""

from __future__ import annotations

import math


def is_prime(number: int) -> bool:
    """Return whether *number* is prime in O(sqrt(n)) time."""

    if number < 2:
        return False
    if number == 2:
        return True
    if number % 2 == 0:
        return False

    limit = math.isqrt(number)
    return all(number % divisor != 0 for divisor in range(3, limit + 1, 2))


def main() -> int:
    """Read one integer, print its classification, and return an exit code."""

    try:
        number = int(input("Enter an integer: ").strip())
    except ValueError:
        print("Error: please enter a valid integer.")
        return 1

    result = "prime" if is_prime(number) else "not prime"
    print(f"{number} is {result}.")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
